tinygrad/runtime/graph/hsa.py

```diff
@@ -61,6 +61,7 @@ def __init__(self, jit_cache: List[JitItem], input_rawbuffers: List[Buffer], var
 
     # Build queues.
     self.virt_aql_queues: Dict[Compiled, VirtAQLQueue] = {dev:VirtAQLQueue(dev, 2*len(self.jit_cache)+16) for dev in self.devices}
+    self.virt_aql_copy_queues: Dict[Compiled, VirtAQLQueue] = {dev:VirtAQLQueue(dev, 2*len(self.jit_cache)+16) for dev in self.devices}
```

Comment:
Why is this in a different queue from the compute kernels?

Reply:
This is faster from my tests. I think because it starts to execute them in parallel. I can try to put them in one queue with some different header and hope that CP will put the in parallel as well.